ui/raidboss/emulator/data/AnalyzedEncounter.js

```diff
@@ -0,0 +1,271 @@
+    }
+
+    let response = {};
+    if (trigger.response) {
+      let r = trigger.response;
+      response = (typeof r === 'function') ? r(data, matches) : r;
+    }
+
+    ret.Response = response;
+
+    // popup text
+    if (typeof trigger.alarmText === 'function' && trigger.alarmText(data, matches))
+      result = trigger.alarmText(data, matches);
+    else if (typeof trigger.alarmText === 'object')
+      result = trigger.alarmText;
+    else if (response.alarmText)
+      result = response.alarmText;
+
+    if (typeof trigger.alertText === 'function' && trigger.alertText(data, matches))
+      result = trigger.alertText(data, matches);
+    else if (typeof trigger.alertText === 'object')
+      result = trigger.alertText;
+    else if (response.alertText)
+      result = response.alertText;
+
+    if (typeof trigger.infoText === 'function' && trigger.infoText(data, matches))
+      result = trigger.infoText(data, matches);
+    else if (typeof trigger.infoText === 'object')
+      result = trigger.infoText;
+    else if (response.infoText)
+      result = response.infoText;
+
+    ret.Result = result;
+
+    // execute run function after text!
+    if (typeof trigger.run === 'function') {
+      run = true;
+      trigger.run(data, matches);
+    }
+
+    // checking for suppression, first local then global
+    if (typeof suppressed[trigger.id] !== 'undefined') {
+      let timestamps = suppressed[trigger.id];
+      for (let i in timestamps) {
+        if (timestamps[i] > timestamp) {
+          ret.Suppressed = timestamps[i];
+          return ret;
+        }
+      }
+    }
+
+    ret.Executed = true;
+
+    if (!result && !run)
+      return ret;
+
+    suppressed[trigger.id] = suppressed[trigger.id] || [];
+
+    if (trigger.suppressSeconds) {
+      if (typeof trigger.suppressSeconds === 'function')
+        timestamp = timestamp + trigger.suppressSeconds(data, matches) * 1000;
+      else
+        timestamp = timestamp + trigger.suppressSeconds * 1000;
+      suppressed[trigger.id].push(timestamp);
+    } else {
+      timestamp = timestamp + suppress * 1000;
+      suppressed[trigger.id].push(timestamp);
+    }
+
+    return ret;
+  }
+
+  static cloneData(data, full = false) {
+    let ret = {};
+    // Use our own extend logic for the top level to avoid cloning the `options` property
+    // This cut the execution time of this code from 41,000ms to 50ms when parsing a 12 minute pull
+    for (let i in data) {
+      if (!full && (i === 'options' || i === 'party'))
+        continue;
+
+      if (typeof data[i] === 'object') {
+        // Potential future bug: jQuery's extend doesn't include
+        // properties with a value of `undefined`
+        if (Array.isArray(data[i]))
+          ret[i] = jQuery.extend(true, [], data[i]);
+        else
+          ret[i] = jQuery.extend(true, {}, data[i]);
+      } else {
+        ret[i] = data[i];
+      }
+    }
+    return ret;
+  }
+
+  JobToJobEnum(Job) {
```

Comment:
Maybe this should live next to `Util.jobEnumToJob`?

Reply:
Probably. I was avoiding making changes to the existing files. I'll move this over.